### PR TITLE
Adding profile abut filestore with xpn

### PR DESCRIPTION
**Pull request by chao007**
Main Update:
1. Add filestore csi driver to job `periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14`
2. Update the storageclass filestore-csi as non default for all qe jobs which including filestore
3. Update scripts `storage-create-csi-gcp-filestore-commands.sh` due to could not work in the proxy env.

### CODE REVIEW

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-gcp-ipi-user-labels-tags-filestore-csi-tp-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-gcp-ipi-user-labels-tags-filestore-csi-tp-arm-f14

**Comment by chao007** [comment]
@jianlinliu @jianli-wei Could you help to review this pr?

File: ci-operator/step-registry/storage/conf/csi-optional/gcp-filestore/networkconf/storage-conf-csi-optional-gcp-filestore-networkconf-commands.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+touch "${SHARED_DIR}/filestore_csi_networkconf.txt"
```
Comment (jianlinliu) [inline ci-operator/step-registry/storage/conf/csi-optional/gcp-filestore/networkconf/storage-conf-csi-optional-gcp-filestore-networkconf-commands.sh:6]: This is only used for a flag to identify it is xpn install, right? If that, I think we can directly use `${SHARED_DIR}/xpn.json` to identify that, not need to introduce this new step.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml [hunk 2]
```diff
@@ -1344,19 +1344,21 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-gcp-ipi-xpn-private
-- as: gcp-ipi-xpn-private-arm-f14
+- as: gcp-ipi-xpn-private-filestore-csi-arm-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml:1347]: Suggest to sync the same change to `gcp-ipi-xpn-private-amd-f28-destructive` too.

File: ci-operator/step-registry/storage/conf/csi-optional/gcp-filestore/networkconf/storage-conf-csi-optional-gcp-filestore-networkconf-commands.sh (hunk 1 above)
Comment (chao007) [reply ci-operator/step-registry/storage/conf/csi-optional/gcp-filestore/networkconf/storage-conf-csi-optional-gcp-filestore-networkconf-commands.sh:6]: Thanks, let me have a try.

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-user-labels-tags-filestore-csi-tp-amd-f28-destructive

File: ci-operator/step-registry/storage/create/csi-gcp-filestore/storage-create-csi-gcp-filestore-commands.sh [hunk 3]
```diff
@@ -32,10 +45,16 @@ provisioner: filestore.csi.storage.gke.io
...
 allowVolumeExpansion: true
 parameters:
+  connect-mode: DIRECT_PEERING
   network: $NETWORK_NAME
   labels: kubernetes-io-cluster-$CLUSTER_NAME=owned
 EOF
 
+if [[ -s "${SHARED_DIR}/xpn.json" ]]
+then
+  sed 's/DIRECT_PEERING/PRIVATE_SERVICE_ACCESS/' $STORAGECLASS_LOCATION >> $STORAGECLASS_LOCATION
```
Comment (jianlinliu) [inline ci-operator/step-registry/storage/create/csi-gcp-filestore/storage-create-csi-gcp-filestore-commands.sh:55]:
```suggestion
  sed -i 's/DIRECT_PEERING/PRIVATE_SERVICE_ACCESS/' $STORAGECLASS_LOCATION
```

**Comment by chao007** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-xpn-private-filestore-csi-arm-f14

**Comment by jianlinliu** [comment]
cc @liangxia to review

**Comment by jianlinliu** [comment]
@chao007 beside QE's approval, this PR also need storage dev team approval

**Comment by chao007** [comment]
@bertinatto Could you help to review this pr?
1. Add sharedvpc parameters in storageclass 
2. Update scripts storage-create-csi-gcp-filestore-commands.sh due to could not work in the proxy env.

**Pull request merged**